qa/rpc-tests/mempool_size_limit.py

```diff
@@ -66,60 +73,128 @@ def load_from_storage(path):
+    outputs = {node.getnewaddress(): Decimal((input_t['amount'] - fee)/NUM_OUT) for i in range(NUM_OUT)}
+    rawtx = node.createrawtransaction(inputs, outputs)
+
+    # build an object from the raw Tx in order to be able to modify it
+    tx_01 = CTransaction()
+    f = BytesIO(unhexlify(rawtx))
+    tx_01.deserialize(f)
+
+    for vout_idx in range(len(tx_01.vout)):
+        tx_01.vout[vout_idx].scriptPubKey = txscript
+
+    tx_01.rehash()
+    signresult = node.signrawtransaction(ToHex(tx_01), inputs, None, "NONE")
+    tsize = len(signresult['hex']) // 2
+    feerate = fee / tsize
+    write_to_file(tmpdir + "/tx" + str(i), feerate, signresult["hex"])
+    tx = {'feerate': feerate, 'hex': signresult["hex"]}
+    return tx, tsize
+'''
+
+def create_big_transaction(node, tmpdir, input_t, script_size, fee, i):
+    assert(fee < input_t['amount'])
+    txscript = create_tx_script(script_size, input_t['amount'] - fee)
+
+    inputs = []
+    inputs.append({ "txid" : input_t["txid"], "vout" : input_t["vout"], "scriptPubKey": input_t["scriptPubKey"]})
     addr = node.getnewaddress()
+    outputs = {addr: Decimal(0)} # This will be overwritten anyway
+    rawtx = node.createrawtransaction(inputs, outputs)
+    out_num_pos = rawtx.find('ffffffff')
+    script_position = rawtx.find('76a914')
+    script_length = int(rawtx[script_position - 2 : script_position], 16) * 2
+
+    newtx = rawtx[0:out_num_pos + 8]
+    newtx = newtx + txscript
+    newtx = newtx + rawtx[script_position + script_length:]
+
+    signresult = node.signrawtransaction(newtx, inputs, None, "NONE")
+    tsize = len(signresult['hex']) // 2
+    feerate = fee / tsize
+    if (tmpdir is not None):
+        write_to_file(tmpdir + "/tx" + str(i), feerate, signresult["hex"])
+    tx = {'feerate': feerate, 'hex': signresult["hex"]}
+    return tx
+
+
+def create_big_transactions(node, tmpdir, utxos, size):
     txs = []
     tot_size = 0
     i = 0
     os.makedirs(tmpdir + "/txs")
+    minfeerate, maxfeerate = MAX_FEE, Decimal("0")
+    fee = Decimal(0.5) + Decimal(0.01) * i
     while (tot_size < size):
-        t = utxos.pop()
-        inputs = []
-        inputs.append({ "txid" : t["txid"], "vout" : t["vout"]})
-        outputs = {addr: satoshi_round(t['amount'])} # This will be overwritten anyway
-        rawtx = node.createrawtransaction(inputs, outputs)
-        out_num_pos = rawtx.find('ffffffff')
-        script_position = rawtx.find('76a914')
-        script_length = int(rawtx[script_position - 2 : script_position], 16) * 2
-
-        newtx = rawtx[0:out_num_pos + 8]
-        newtx = newtx + txscript
-        newtx = newtx + rawtx[script_position + script_length:]
-
-        signresult = node.signrawtransaction(newtx, None, None, "NONE")
-        tsize = len(signresult['hex']) // 2
+        tx = create_big_transaction(node, tmpdir + "/txs", utxos.pop(), 512 * NUM_OUT, fee, i)
         tot_size += tsize
```

Comment:
...or let `create_big_transaction()` return `tsize` as well, as it is on `create_big_transaction_noraw()`

Reply:
Yes, this was the intended implementation, thank you